galley/formatted_queries.py

```diff
@@ -17,6 +17,21 @@
 logger = logging.getLogger(__name__)
 
 
+def get_external_name_for_ingredient(ingredient: Dict) -> Optional[str]:
+    name = (ingredient.get("externalName") or ingredient.get("name"))
+    ingredientListStr = (get_primary_vendor_item_for_ingredient(ingredient) or {}).get("ingredientListStr")
+    return f"{name} ({ingredientListStr})" if name and ingredientListStr else name
+
+
+def get_primary_vendor_item_for_ingredient(ingredient: Dict) -> Optional[Dict]:
+    locationVendorItems: Dict = next(iter(ingredient.get('locationVendorItems',[])),{})
```

Comment:
NB: This will error if `locationVendorItems` is `None`.  If it's possible for this property to be `None` we might want to consider the following instead.
```suggestion
    locationVendorItems: Dict = next(iter(ingredient.get('locationVendorItems') or []),{})
```
This will handle none exiting property as well as None. It won't handle a value of the wrong type (not a List), but arguable that should be an exception (or we should used typed objects rather than dicts)

Reply:
added the `or`, thanks for the suggestion